linera-service/src/linera/main.rs

```diff
@@ -1448,6 +1459,110 @@ impl ClientOptions {
     }
 }
 
+struct DatabaseToolJob<'a>(&'a DatabaseToolCommand);
+
+#[async_trait]
+impl RunnableWithStore for DatabaseToolJob<'_> {
+    type Output = i32;
+
+    async fn run<S>(
+        self,
+        config: S::Config,
+        namespace: String,
+    ) -> Result<Self::Output, anyhow::Error>
+    where
+        S: KeyValueStore + Clone + Send + Sync + 'static,
+        S::Error: Send + Sync,
+    {
+        match self.0 {
+            DatabaseToolCommand::DeleteAll => {
+                let start_time = Instant::now();
```

Comment:
(This could be deduplicated. But it looks like you moved it into the `match` arms on purpose?)